python/sparkdl/param/converters.py

```diff
@@ -0,0 +1,155 @@
+        """
+        Convert a value to a :py:obj:`tf.Tensor` name to column name mapping
+        as a sorted list of string pairs, if possible.
+        """
+        if isinstance(value, dict):
+            strs_pair_seq = []
+            for _maybe_tnsr_name, _maybe_col_name in value.items():
+                # Check if the non-tensor value is of string type
+                _col_name = _get_strict_col_name(_maybe_col_name)
+                # Check if the tensor name is actually valid
+                _tnsr_name = _get_strict_tensor_name(_maybe_tnsr_name)
+                strs_pair_seq.append((_tnsr_name, _col_name))
+
+            return sorted(strs_pair_seq)
+
+        err_msg = "Could not convert [type {}] {} to tf.Tensor name to column name mapping"
+        raise TypeError(err_msg.format(type(value), value))
+
+    @staticmethod
+    def toTFHParams(value):
+        """ Convert a value to a :py:obj:`tf.contrib.training.HParams` object, if possible. """
+        if isinstance(value, tf.contrib.training.HParams):
+            return value
+        else:
+            raise TypeError("Could not convert %s to TensorFlow HParams" % type(value))
+
+    @staticmethod
+    def toStringOrTFTensor(value):
+        """ Convert a value to a str or a :py:obj:`tf.Tensor` object, if possible. """
+        if isinstance(value, tf.Tensor):
+            return value
+        try:
+            return TypeConverters.toString(value)
+        except Exception as exc:
+            err_msg = "Could not convert [type {}] {} to tf.Tensor or str. {}"
+            raise TypeError(err_msg.format(type(value), value, exc))
+
+    @staticmethod
+    def supportedNameConverter(supportedList):
+        """
+        Create a converter that try to check if a value is part of the supported list.
+
+        :param supportedList: list, containing supported objects.
+        :return: a converter that try to convert a value if it is part of the `supportedList`.
+        """
+        def converter(value):
+            if value in supportedList:
+                return value
+            err_msg = "[type {}] {} is not in the supported list: {}"
+            raise TypeError(err_msg.format(type(value), str(value), supportedList))
+
+        return converter
+
+    @staticmethod
+    def toKerasLoss(value):
+        """ Convert a value to a name of Keras loss function, if possible """
+        if kmutil.is_valid_loss_function(value):
+            return value
+        err_msg = "Named loss not supported in Keras: [type {}] {}"
+        raise ValueError(err_msg.format(type(value), value))
+
+    @staticmethod
+    def toKerasOptimizer(value):
+        """ Convert a value to a name of Keras optimizer, if possible """
+        if kmutil.is_valid_optimizer(value):
+            return value
+        err_msg = "Named optimizer not supported in Keras: [type {}] {}"
+        raise TypeError(err_msg.format(type(value), value))
+
+
+def _get_strict_tensor_name(_maybe_tnsr_name):
```

Comment:
As per my comment above, this can be changed to `_check_is_tensor_name(name)` that just checks that name is a string (using `_check_is_str`) and that doesn't have a ":".

The current logic in `_get_strict_tensor_name` duplicates checks in `as_tensor_name`, and `as_tensor_name` does more than what we want here, so there is no need to call that. I understand we want to potentially give more descriptive error messages, but I think just checking that it roughly looks like a tensor name is good enough (and that's all we're doing here currently anyway).